submariner/IPV6-datapath.md

```diff
@@ -0,0 +1,185 @@
+# Submariner Enhancement for IPV6 datapath
+
+<!-- Add link to issue/epic if available -->
+
+## Summary
+
+IPv4, IPV6 and dual-stack networking is supported for Kubernetes cluster starting in 1.21.
+IPV6 networking allowing assignment of IPv6 addresses.
+Dual-stack networking allowing the simultaneous assignment of both IPv4 and IPv6 addresses.
+
+IPv4/IPv6 dual-stack on your Kubernetes cluster provides the following features:
+
+* Dual-stack Pod networking (a single IPv4 and IPv6 address assignment per Pod)  
+* IPv4 and IPv6 enabled Services  
+* Pod off-cluster egress routing (eg. the Internet) via both IPv4 and IPv6 interfaces
+
+Currently, Submariner only supports IPV4 datapath and this proposal explains the changes required to support IPV6 and dual-stack clusters.
+
+## Overview
+
+Submariner’s control plane uses a central broker component to facilitate the exchange of metadata information between participating clusters.
+
+Both inter-cluster datapath and service-discovery (Lighthouse) apply their configuration based on the information received from control plane.
+
+The Broker, inter-cluster datapath and Lighthouse should be updated to support mixed clusters environment (e.g: dual-stack, V6 only , V4 only).
+
+### Control plane
+
+For mixed clusters environment we may experience connectivity issue.
+Some clusters and broker may not connect if the Broker’s cluster doesn’t support dual-stack.
+A dual-stack cluster should be selected as the broker in this case.
+
+### Inter-cluster datapath
+
+The inter-cluster datapath needs to support clusters with different networking configuration as described in the next table:
+
+| clusterA networking | clusterB networking | Supported connectivity type |
+| :---- | :---- | :---- |
+| V4 | V4 | V4 |
+| V4 | V6 | N/A |
+| V6 | dual-stack | V6 |
+| V6 | V6 | V6 |
+| V4 | dual-stack | V4 |
+| dual-stack | dual-stack | V4,V6 |
+
+### Service Discovery
+
+Lighthouse needs to handle imported services with a different IP family than the local cluster.
+This is necessary in a mixed cluster environment, as described below.
+
+|  | Local cluster networking | Imported service networking | Supported DNS record  |
+| :---- | :---- | :---- | :---- |
+| 1 | V4 | V4 | V4 |
+| 2 | V4 | V6 | N/A |
+| 3 | V6 | dual-stack | V6 |
+| 4 | V6 | V6 | V6 |
+| 5 | V4 | dual-stack | V4 |
+| 6 | dual-stack | dual-stack | V4,V6 |
+
+This can be achieved in one of the following ways:
+
+* Lighthouse ignores processing imported services IP addresses that don’t match local cluster networking configuration.
+* Lighthouse will process imported services regardless of local cluster networking configuration, and rely on local workloads dns requests.
```

Comment:
Looks like these are contradicting statements. Can you please rephrase.

Reply:
I've rephrased it, let me know if it looks OK to you.